packages/compass-connection-import-export/src/hooks/use-export.ts

```diff
@@ -59,6 +60,11 @@ export function useExportConnections(
   onChangeRemoveSecrets: (evt: React.ChangeEvent<HTMLInputElement>) => void;
   state: ExportConnectionsState;
 } {
+  const connectionStorage = getCompassRendererConnectionStorage();
+  const exportConnectionsMethod: CompassConnectionStorage['exportConnections'] =
+    exportConnections ??
+    connectionStorage.exportConnections.bind(connectionStorage);
```

Comment:
So while we know that this is compass-only, to make sure all our code is actually pulling the implementations from the same place and follows the same patterns can we use a hook here instead? Even if it means that we'll be doing something like

```
const connectionStorage = useConnectionStorage()
const exportConnections = __TEST_EXPORT_CONNECTIONS ?? connectionStorage.exportConnections?.bind(connectionStorage) ?? () => { throw new Error('Export Connections feature requires Connection Storage with implemented exportConnections method'); }
```

If we ensure that all plugins are pulling the services from the correct place, we then don't even need to have this special singleton get method that you added, we can just set it up once in compass home, as providers is what guarantees that all code is using the same instance

Reply:
Thank you for this. We don't have the singleton anymore and instead we always use the injected service.